crates/runtime/src/accelerated_table.rs

```diff
@@ -434,18 +452,29 @@ impl AcceleratedTable {
                     }
                 }
                 AccelerationRefreshMode::Full(receiver) => {
+
+                    let refresh_lock = refresh.read().await;
```

Comment:
We can be slightly more efficient with how we're taking the locks here to avoid some clones:

```rust
use std::ops::Deref;
// ...
                AccelerationRefreshMode::Full(receiver) => {

                    let refresh_lock = refresh.read().await;
                    let refresh_settings = refresh_lock.deref();

                    let schema = federated.schema();
                    let column = refresh_settings.time_column.as_deref().unwrap_or_default();
                    let field = schema.column_with_name(column).map(|(_, f)| f);
                    let filter_converter = TimestampFilterConvert::create(field, refresh_settings.time_column.clone(), refresh_settings.time_format.clone());
                    drop(refresh_lock);

                    let mut refresh_stream = ReceiverStream::new(receiver);

                    while refresh_stream.next().await.is_some() {
                        tracing::info!("[refresh] Loading data for dataset {dataset_name}");
                        status::update_dataset(&dataset_name, status::ComponentStatus::Refreshing);

                        let refresh_lock = refresh.read().await;
                        let refresh_settings = refresh_lock.deref();

                        let timer = TimeMeasurement::new("load_dataset_duration_ms", vec![("dataset", dataset_name.clone())]);
                        let filters = match (refresh_settings.period, filter_converter.as_ref()){
                            (Some(period), Some(converter)) => {
                                let start = SystemTime::now() - period;

                                let Ok(timestamp) = get_timestamp(start) else {
                                    tracing::error!("[refresh] Failed to calculate timestamp of refresh period start");
                                    continue;
                                };
                                vec![converter.convert(timestamp, Operator::Gt)]
                            },
                            _ => vec![],
                        };

                        let refresh_sql = refresh_settings.sql.clone();
                        drop(refresh_lock);

                        let data = match get_data(&mut ctx, OwnedTableReference::bare(dataset_name.clone()), Arc::clone(&federated), refresh_sql, filters).await {
```

Reply:
Thank you!